### ISSUE TITLE
 Use Mesos task labels to tag requests

### ISSUE THREAD

**Issue by apognu** [HubSpot/Singularity#823]
Mesos 0.22 introduced task labels as a way of adding metadata to tasks.

It would be useful to be able to tag the requests in Singularity (and therefore the tasks that are spawned from its deploys) during their creation (and even afterwards). Possible use-cases would be monitoring, load-balancing or any external system that could need to select tasks by arbitrary metadata.

As far as I know, the only way to do that with Singularity would be to encode those metadata into the request name, making for rigid (and long) IDs.

Is such a feature planned or being discussed?

Regards.

**Comment by tpetr** [comment]
We already support setting labels in the API (via #657 -- [the `labels` field in `SingularityDeploy`](https://github.com/HubSpot/Singularity/blob/master/SingularityBase/src/main/java/com/hubspot/singularity/SingularityDeploy.java#L44)), but labels currently do not affect task scheduling decisions or are surfaced in the UI. We can look into adding them.

**Comment by apognu** [comment]
Oh, I did miss that in the docs.

However, when I set my labels with the following:

```
{
  "deploy": {
    "requestId": "demo-rec",
    "id": "4",
    "arguments": [
    ],
    "env": {
      "ENV": "staging",
      "PORT": "3000"
    },
    "containerInfo": {
      "type": "DOCKER",
      "docker": {
        "image": "my/image:latest",
        "privileged": false,
        "network": "BRIDGE",
        "portMappings": [
          {
            "containerPortType": "LITERAL",
            "containerPort": 3000,
            "hostPortType": "FROM_OFFER",
            "hostPort": 0
          }
        ]
      }
    },
    "resources": {
      "cpus": 0.1,
      "memoryMb": 16,
      "numPorts": 1
    },
    "uris": [
      "file:///etc/docker/repo.conf"
    ],
    "labels": {
      "lorem": "ipsum",
      "tags": "one,two,three"
    }
  }
}
```

The mesos task does not seem to be populated with them, this is what I get from ```/master/state.json```:

```
"labels": [
[...]
  {
    "key": "Docker.NetworkSettings.IPAddress",
    "value": "172.17.0.3"
  }
],
[...]
```

Am I setting the labels properly in my deploy definition?

**Comment by apognu** [comment]
I'm guessing I am, since the labels are reported back in Singularity API on that request's active deploy.

Are Singularity's labels not designed to bubble up all the way to the corresponding Mesos task?

**Comment by ssalinas** [comment]
Hey, thanks for pointing this out. I had honestly thought I wired this up when we moved to 0.23, but I guess I missed it. Fixed in #824 once I test and get it merged in

**Comment by tpetr** [comment]
@apognu task label fix has been merged to master
